lambda_strip_docx/test_handler.py

```diff
@@ -591,40 +573,42 @@ def test_lambda_handler_successful_processing(self, s3_with_docx_file, input_byt
         # Call lambda handler
         lambda_handler(event, {})
 
-        # Verify the processed file was uploaded
-        processed_key = "sample_with_author_processed.docx"
-
-        # Check that processed file exists
+        # Verify the file was processed in place (same object key)
+        # Check that the original file still exists
         response = s3_client.list_objects_v2(Bucket=bucket_name)
         object_keys = [obj['Key'] for obj in response.get('Contents', [])]
-        assert processed_key in object_keys, f"Processed file {processed_key} not found in S3"
+        assert object_key in object_keys, f"Processed file {object_key} not found in S3"
 
         # Get the processed file and verify it's different from original
-        processed_response = s3_client.get_object(Bucket=bucket_name, Key=processed_key)
+        processed_response = s3_client.get_object(Bucket=bucket_name, Key=object_key)
         processed_content = processed_response['Body'].read()
-
         assert processed_content != input_bytes, "Processed content should be different from input"
         assert len(processed_content) > 0, "Processed content should not be empty"
 
     def test_lambda_handler_skips_non_docx_files(self, s3_setup):
         """Test lambda handler skips non-DOCX files"""
         s3_client, bucket_name = s3_setup
+        original_content = b"text file content"
 
         # Upload a non-DOCX file
         non_docx_key = "test.txt"
-        s3_client.put_object(Bucket=bucket_name, Key=non_docx_key, Body=b"text file content")
+        s3_client.put_object(Bucket=bucket_name, Key=non_docx_key, Body=original_content)
 
         # Create S3 event for the non-DOCX file
         event = create_s3_event(bucket_name=bucket_name, object_key=non_docx_key)
 
         # Call lambda handler
         lambda_handler(event, {})
 
-        # Verify no processed file was created
-        response = s3_client.list_objects_v2(Bucket=bucket_name)
-        object_keys = [obj['Key'] for obj in response.get('Contents', [])]
-        processed_keys = [key for key in object_keys if "_processed" in key]
-        assert len(processed_keys) == 0, "No processed files should be created for non-DOCX files"
+        # Verify the file was not modified (content should remain the same)
+        response = s3_client.get_object(Bucket=bucket_name, Key=non_docx_key)
+        current_content = response['Body'].read()
+        assert current_content == original_content
+
+        # Verify no version tag was added
+        tag_response = s3_client.get_object_tagging(Bucket=bucket_name, Key=non_docx_key)
+        tags = {tag['Key']: tag['Value'] for tag in tag_response.get('TagSet', [])}
+        assert 'DOCUMENT_PROCESSOR_VERSION' not in tags
```

Comment:
Should this also raise an error?

Reply:
as in, should we change the lambda to raise an error if the event passed in includes a non docx file? I thought just logging it and skipping it was sufficient but happy to discuss